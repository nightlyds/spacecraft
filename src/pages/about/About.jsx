import React, { useState } from "react";
import counterpart from "counterpart";
import "./css/about.css";
import "./css/media_about.css";
import en from "./lang/en";
import uk from "./lang/uk";
import MenuButton from "../../basic/components/menu/MenuButton.jsx";
import Menu from "../../basic/components/menu/Menu.jsx";
import Footer from "../../basic/components/Footer.jsx";
import LanguageChange from "../../basic/components/LanguageChange.jsx";
import Info from './components/info/Info.jsx'

counterpart.registerTranslations("en", en);
counterpart.registerTranslations("uk", uk);
counterpart.setLocale("en");

const About = () => {
  let [menuOpenCheck, setMenuOpen] = useState(false);

  function menuOpen() {
    setMenuOpen(!menuOpenCheck);
  }
  return (
    <div className="about">
      <MenuButton menuOpen={menuOpen} icon={menuOpenCheck} />
      {menuOpenCheck && <Menu />}
      <LanguageChange />
      <div className="about-content">
        <Info />
      </div>
      <Footer />
    </div>
  );
};

export default About;
