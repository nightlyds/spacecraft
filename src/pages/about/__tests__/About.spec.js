import React from 'react';
import About from '../About.jsx'
import shallow from 'enzyme/build/shallow';

const setUp = (props) => shallow(<About {...props} />)

describe("About component tests", () => {
    let component;

    beforeEach(() => {
        component = setUp();
    })

    it("Snapshot", () => {
        expect(component).toMatchSnapshot();
    })
})