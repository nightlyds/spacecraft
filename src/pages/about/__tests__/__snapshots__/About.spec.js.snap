// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component tests Snapshot 1`] = `
<div
  className="about"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="about-content"
  >
    <Info />
  </div>
  <Footer />
</div>
`;
